### PR TITLE
Bridge info settings tab

### PR DESCRIPTION
**Pull request by Half-Shot**
This is a room settings tab that shows the connected bridges in a room, which is my proof of concept on the client side for [MSC2346](https://github.com/matrix-org/matrix-doc/pull/2346).

image

**EDIT**: Added a few more details to the page

**Paired with https://github.com/vector-im/riot-web/pull/11778** - TR

### CODE REVIEW

**Comment by Half-Shot** [comment]
Notable changes from today:
- i18n is now being used
- Removed `active` status, since it's been removed from the MSC.

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2019 New Vector Ltd
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: We assign all copyright to the foundation:
```suggestion
Copyright 2019 The Matrix.org Foundation C.I.C.
```

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    "uk.half-shot.bridge",
+    // m.bridge
+];
+
+export default class BridgeSettingsTab extends React.Component {
+    static propTypes = {
+        roomId: PropTypes.string.isRequired,
+    };
+
+    constructor() {
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: this constructor can be removed

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        super();
+
+        this.state = {
+        };
+    }
+
+    componentWillMount() {
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: This function can be removed

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        let creator = null;
+        if (content.creator) {
+            const pill = <Pill
+                type={Pill.TYPE_USER_MENTION}
+                room={room}
+                url={makeUserPermalink(content.creator)}
+                shouldShowPillAvatar={true}
+            />;
+            creator = (<p>{
+                _t("This bridge was provisioned by %(pill)s", {
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: This should be an HTML replacement instead, also with a descriptive name for our translation community:
```suggestion
                _t("This bridge was provisioned by <user />", {
```

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                    pill,
+                })
+            } </p>);
+        }
+
+        const bot = (<p> {_t("This bridge is managed by the %(pill)s bot user.", {
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: same thing about html

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+        let creator = null;
+        if (content.creator) {
+            const pill = <Pill
+                type={Pill.TYPE_USER_MENTION}
+                room={room}
+                url={makeUserPermalink(content.creator)}
+                shouldShowPillAvatar={true}
+            />;
+            creator = (<p>{
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: If you're using parens then the jsx should be on the next line

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+            pill: <Pill
+                type={Pill.TYPE_USER_MENTION}
+                room={room}
+                url={makeUserPermalink(event.getSender())}
+                shouldShowPillAvatar={true}
+                />,
+        })} </p>);
+        let channelLink = channelName;
+        if (channel.external_url) {
+            channelLink = <a target="_blank" href={channel.external_url}>{channelName}</a>;
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]:
```suggestion
            channelLink = <a target="_blank" href={channel.external_url} rel="noopener">{channelName}</a>;
```

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+        }
+
+        let networkLink = networkName;
+        if (network && network.external_url) {
+            networkLink = <a target="_blank" href={network.external_url}>{networkName}</a>;
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]:
```suggestion
            networkLink = <a target="_blank" href={network.external_url} rel="noopener">{networkName}</a>;
```

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+        }
+
+        const chanAndNetworkInfo = (
+            (_t("Bridged into %(channelLink)s %(networkLink)s, on %(protocolName)s", {
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: Parens before the translation aren't required, including the wrapping ones.

Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: also html

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+        );
+
+        let networkIcon = null;
+        if (networkName && network.avatar) {
+            const avatarUrl = ContentRepo.getHttpUriForMxc(
+                MatrixClientPeg.get().getHomeserverUrl(),
+                network.avatar, 32, 32, "crop",
+            );
+            networkIcon = <BaseAvatar width={32} height={32} resizeMethod='crop'
+                            name={ networkName } idName={ networkName }
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: It's super annoying imo, but we've decided that properties are to be indented 4 spaces, not aligned.

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+                            url={ avatarUrl } />;
+        }
+
+        let channelIcon = null;
+        if (channel.avatar) {
+            const avatarUrl = ContentRepo.getHttpUriForMxc(
+                MatrixClientPeg.get().getHomeserverUrl(),
+                channel.avatar, 32, 32, "crop",
+            );
+            console.log(channel.avatar);
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: this should be removed

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+            channelIcon = <BaseAvatar width={32} height={32} resizeMethod='crop'
+                            name={ networkName } idName={ networkName }
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: 4 spaces

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+                            url={ avatarUrl } />;
+        }
+
+        const heading = _t("Connected to %(channelIcon)s %(channelName)s on %(networkIcon)s %(networkName)s", {
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: html

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+            channelIcon,
+            channelName,
+            networkName,
+            networkIcon,
+        });
+
+        return (<li key={event.stateKey}>
+            <div>
+                <h3> {heading} </h3>
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: I don't think the spaces around this are helping you with anything?

**Comment by Half-Shot** [comment]
Note that in https://github.com/matrix-org/matrix-react-sdk/pull/3693/commits/50e19ba43d64e168e670e4657153f9359477bcb6 I made the change to remove "bot user" from the text. Due to matrix being how it is, we cannot determine if the sender of the state event is a bot user.

To err on the side of paranoia, let's not imply they are a bot user in the text to avoid any weirdness. Users should hopefully be able to distinguish a user from a bot by the displayname/mxid, or at least not be intentionally misled.

File: src/components/views/dialogs/RoomSettingsDialog.js [hunk 15]
```diff
@@ -73,6 +78,15 @@ export default class RoomSettingsDialog extends React.Component {
             "mx_RoomSettingsDialog_rolesIcon",
             <NotificationSettingsTab roomId={this.props.roomId} />,
         ));
+
+        if (shouldShowBridgeIcon) {
+            tabs.push(new Tab(
+                _td("Bridge Info"),
```
Comment (Half-Shot) [inline src/components/views/dialogs/RoomSettingsDialog.js]: I wonder if this should just be "Bridges", or "Connected Bridges", or "Integrations"

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 16]
```diff
@@ -101,9 +87,13 @@ export default class BridgeSettingsTab extends React.Component {
                 MatrixClientPeg.get().getHomeserverUrl(),
                 network.avatar, 32, 32, "crop",
             );
-            networkIcon = <BaseAvatar width={32} height={32} resizeMethod='crop'
-                            name={ networkName } idName={ networkName }
-                            url={ avatarUrl } />;
+            networkIcon = <BaseAvatar
+                              width={32}
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: we indent by 4, not 18

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 17]
```diff
@@ -112,13 +102,16 @@ export default class BridgeSettingsTab extends React.Component {
                 MatrixClientPeg.get().getHomeserverUrl(),
                 channel.avatar, 32, 32, "crop",
             );
-            console.log(channel.avatar);
-            channelIcon = <BaseAvatar width={32} height={32} resizeMethod='crop'
-                            name={ networkName } idName={ networkName }
-                            url={ avatarUrl } />;
+            channelIcon = <BaseAvatar
+                              width={32}
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: indentation

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+        );
+
+        let networkIcon = null;
+        if (networkName && network.avatar) {
+            const avatarUrl = ContentRepo.getHttpUriForMxc(
+                MatrixClientPeg.get().getHomeserverUrl(),
+                network.avatar, 32, 32, "crop",
+            );
+            networkIcon = <BaseAvatar
+    width={32}
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: It seems like you're taking the rules a bit too literally ._.

I'm aware of how silly the linting rules are, but the intention is to have 4 spaces of indentation from the logical point:

```
            networkIcon = <BaseAvatar
                width={32}
                height={32}
                etc..
            />;
```

Comment (Half-Shot) [reply src/components/views/settings/tabs/room/BridgeSettingsTab.js]: Err, right. It did look a bit weird...

**Comment by Half-Shot** [comment]
@turt2live we good now?

**Pull request merged**